### DISCUSSION TITLE
Automatic determination of (RA,Dec) full sky coordinates at pixel-level for allsky images

### DISCUSSION THREAD

**Discussion by gaitskell** [aaronwmorris/indi-allsky#1204] [Ideas]
Has anyone had success creating a pipeline which automatically determines the full sky coordinates for each allsky image so any desired object can then be located within a series of stored images over many months?

Have you found it possible to determine a near pixel level accurate (RA,Dec) coordinate mapping over the full image? Was this done with manual identification of reference stars or have you been able to automate it?

I have changed the mounting setup for my camera enough times that automating coordinate determination would be helpful when going back over multiple months of images.

**Comment by mburge** [comment]
Unfortunately I'm not here to say I have a solution. I took a quick stab at doing this, and I can outline the approaches, but the end result was that automated plate solving for arbitrary wide-field images without a prior knowledge is challenging.

A quick manual solution is to take one of your good images, use astrometric to solve it, and then enter the coordinates in the FITS header using config.

You could automate this, the package is available to install locally but it isn't a clean solution. At least I didn't feel it was.

**Comment by bluthen** [comment]
Fish eye lenses have a lot of distortion. I think you would have to plate solve sections of the image, or if you plate solve a center crop and apply a lens distortion profile. When I've tried, the farther from center of the image the harder it was to platesolve that section. It would be a nice if there was a utility that is a fisheye plate solver that takes all that into account.

**Comment by gaitskell** [comment]
@mburge thank you so much for your guidance.  (Understood - as you suggest one can get a pretty good estimate of where the celestial pole and azimuth directions are from a limited number of images. I wanted to then improve on the calibration so that when using the astropy library the Alt Az calculations from code and from the allsky image could be brought more closely into line.)

When you suggest adding the coordinates into FITS header what parameter system did you use for representing the CCD x-y <-> alt az (or zenith) mapping for an allsky image? Is there a standard FITS representation for this? The FITS header contains OBJCTAZ (assume this is center of CCD XY), OBJCTALT (assume orientation of CCD), SCALE (arcsec/pixel,at zenith). I see indi-allsky CONFIG has a specification for ALT & AZ offsets and also some X,Y offsets for Cardinal (I assume these are used if Zenith is not in the center of CCD field, and camera orientation in not N-S?).  I am not sure where the SCALE in FITS is coming from, since it is not the one I calculate for my lens, and I don't know how to edit it in indiallsky.

I understand coordinates for a narrow field frame (with a specific target), but an allsky frame representation has a more complex transforms given non-linear scaling. How does the FITS header information work for this?
